opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -337,12 +335,23 @@ def detect(self) -> "Resource":
                 else sys.version_info,
             )
         )
-
+        _process_pid = os.getpid()
+        _process_executable_name = sys.executable
+        _process_executable_path = os.path.dirname(_process_executable_name)
```

Comment:
We can probably get `process.owner` through `psutil` with `psutil.Process(pid).username`. However this will introduce a third party library. Is this something that we would want to try to do?

@ocelotl @aabmass @srikanthccv  wdyt?

Reply:
Does this detector need to be part of SDK? Can it be an independent detector?

Reply:
I think this detector is part of the SDK, so I don't recommend adding a dependency for psutil.
this is just my opinion.

Reply:
> I think this detector is part of the SDK

Where does it say that process resource detector should be part of SDK?

Reply:
Oh, right, process runtime resources was removed from the Open Telemetry Specification v1.26.0!

I checked and it looks like [resource semantic transformation for processes mentioned in that PR](https://github.com/open-telemetry/opentelemetry-python/pull/2660) has been removed, should i remove this PR?

Note.
- https://github.com/open-telemetry/opentelemetry-python/issues/3468
- https://github.com/open-telemetry/opentelemetry-specification/blob/v1.25.0/specification/resource/semantic_conventions/process.md

Reply:
Wouldn't removing it be a breaking change to the SDK though? @srikanthccv

Reply:
I understand, I'll fix it so that `psutil` can only get the process username when it can import.

Reply:
I think this PR is fine, so I'll add something to get the process.owner value from this PR using psutil?
@lzchen